subjects/traversal.py

```diff
@@ -20,7 +21,7 @@ def rebuild_tree(self, subject, parent, left):
             p.rght = right
             try:
                 p.save()
-            except Exception:
+            except IntegrityError:
```

Comment:
Can you provide a reference for this exception being the one thrown by the save method?

Reply:
I ask because it seems to me like it could also possibly raise a ValueError: https://docs.djangoproject.com/en/1.11/_modules/django/db/models/base/#Model.save

Reply:
Those are references to the exception itself, but how did you determine that that is the exception that needed to be caught? The link I've posted shows the actual source code for the save method, and it looks like it can raise a ValueError in several places. The IntegrityError is also mentioned in a comment, but I'm not clear on when it gets raised. I'm not suggesting we _shouldn't_ catch it, I just think we _also_ need to catch the ValueError. I'm also curious about what led you to catch that one, since I think it's something I myself might have unfortunately missed

Reply:
I thought by updating the child nodes, if it is not satisfying the DB design requirements like the data types or primary key, foreign key references this would be an appropriate exception to catch. And moreover i tried to look for database related exception since the code was updating the child nodes into the database. 
https://docs.djangoproject.com/en/2.1/ref/exceptions/

Reply:
Hmmmm, well I get where you're coming from, but I also think that it might be possible for some of those other database errors to get raised as well maybe? What do you think about just catching ValueError _and_ the main DatabaseError? DatabaseError is the base class for the IntegrityError and most of the other database exceptions, so it would catch all of those.

Reply:
Also, be careful of which _version_ of the django docs you're looking at, as what you linked is for 2.1, but this particular app is just barely getting to 1.11. There's no appreciable change when it comes to this discussion, but it's something to just be careful of :smile:

Reply:
oh..yeah..My bad!! Will make sure from next time!!

Reply:
no big deal, just wanted to let you know in case you hadn't seen the version selector yet. like I said, doesn't make a difference for the stuff we're talking about right now